compiler/frontend/src/org/jetbrains/kotlin/psi/psiUtil/psiUtils.kt

```diff
@@ -63,6 +63,13 @@ val PsiElement.parentsWithSelf: Sequence<PsiElement>
 val PsiElement.parents: Sequence<PsiElement>
     get() = parentsWithSelf.drop(1)
 
+fun PsiElement.matchParents(vararg parentClasses: Class<out PsiElement>): Boolean {
+    return this.parents.zip(sequenceOf(*parentClasses)).all {
+        val (elem, clazz) = it
+        clazz.isAssignableFrom(elem.javaClass)
```

Comment:
With lambda argument destructuring and `isInstance` method you could write this method as

```
    return this.parents.zip(sequenceOf(*parentClasses)).all { (elem, clazz) ->
        clazz.isInstance(elem)
    }
```